### ISSUE TITLE
child_changed does not work with query

### ISSUE THREAD

**Issue by ewindso** [invertase/react-native-firebase#556]
### Issue

I'm trying to limit the items I get `child_changed` triggers on (initially posted here: https://stackoverflow.com/questions/46897130/firebase-child-changed-with-equalto)

For example:

```
const ref = firebase.database()
                    .ref('images')
                    .orderByValue('place')
                    .equalTo('san-francisco')

ref.on('child_changed', data => {
  console.log(data)
})
```

Somebody recommended `orderByChild`, I tried that and it also did not work.

If I remove `equalTo`, I get updates but they are too broad.

Thank you for your time.

### Environment

1. Application Target Platform:  iOS

2. Development Operating System: macOS Sierra

3. Build Tools: XCode, iOS 11

4. React Native version: 0.46.4

5. RNFirebase Version: 2.2.2

6. Firebase Module: database

**Comment by Salakar** [comment]
@ewindso are you able to query the above in the firebase web sdk with it working?

**Comment by chrisbianca** [comment]
@ewindso There are a lot of fixes in the v3.0.x stream that I imagine will resolve this issue.  Is there a reason you haven't looked to upgrade yet?

**Comment by chrisbianca** [comment]
Closing due to lack of response. Please re-open if you're still experiencing issues and let us know the answers to the above questions.
